perception_dataset/deepen/download_annotations.py

```diff
@@ -2,48 +2,183 @@
+
+        label_response_data = label_response.json()
+        pcd_files = [
+            pcd_file_name
+            for pcd_file_name in label_response_data["labels"].keys()
+            if pcd_file_name.endswith(EXTENSION_ENUM.PCD.value)
+        ]
+
+        paint_metadata = json.loads(response.headers["paint-metadata"])
+        assert len(pcd_files) == len(
+            paint_metadata["frame_sizes"]
+        ), "Number of pcd files and frame sizes do not match!"
+
+        decompressed_data = zlib.decompress(response.content)
+        lidarseg_category_indices = np.frombuffer(decompressed_data, dtype=np.uint8)
+        # Split based on pointcloud size
+        previous_frame_size = 0
+
+        for file_id, frame_size in zip(pcd_files, paint_metadata["frame_sizes"]):
+
+            lidarseg_anno_filename = (
+                dataset_path / f"{dataset_id}_{str(file_id)}{EXTENSION_ENUM.BIN.value}"
+            )
+
+            annos_info = {
+                "dataset_id": dataset_id,
+                "file_id": file_id,
+                "label_id": "none",  # Keep it for consistency with downstream tasks
+                "label_category_id": "none",  # Keep it for consistency with downstream tasks
+                "total_lidar_points": frame_size,
+                "sensor_id": "lidar",
+                "stage_id": "QA",
+                "paint_categories": paint_metadata["paint_categories"],
+                "lidarseg_anno_file": str(lidarseg_anno_filename),
+            }
+
+            with open(lidarseg_anno_filename, "wb") as f:
+                lidarseg_category_indices[
+                    previous_frame_size : previous_frame_size + frame_size
+                ].tofile(f)
+
+            previous_frame_size = frame_size
+            lidarseg_annos_info.append(annos_info)
+
+    # Save lidarseg anno information to json
+    output_name = Path(dataset_dir, output_name)
+    output_name.parents[0].mkdir(parents=True, exist_ok=True)
+    with open(output_name, "w") as f:
+        json.dump(lidarseg_annos_info, f, indent=4)
+
+    print(f"Annotation file is saved: {output_name}")
+
+
+def get_datasets(dataset_ids: List[str], dataset_dir: str, output_name: str, label_type: str):
+    if label_type == "labels":
+        labels_request_func = get_annotation_labels
+    elif label_type == "paint3d":
+        labels_request_func = get_paint3d_labels
+    else:
+        raise ValueError(f"Label type: {label_type} not supported!")
+
+    labels_request_func(dataset_ids=dataset_ids, dataset_dir=dataset_dir, output_name=output_name)
+
+
 if __name__ == "__main__":
     parser = argparse.ArgumentParser()
     parser.add_argument(
         "--config",
         type=str,
         default="config/convert_deepen_to_t4.yaml",
     )
+    parser.add_argument(
+        "--label_type",
+        type=str,
+        default="labels",
+        help="Annotation type to request labels, supported values: ['labels', 'paint3d']",
+    )
```

Comment:
[NITS] `choices` makes clear options that we can choose.
```suggestion
    parser.add_argument(
        "--label_type",
        type=str,
        default="labels",
        choices=["labels", "paint3d"],
        help="Annotation type to request labels, supported values: ['labels', 'paint3d']",
    )
```

Reply:
Yeah, it looks better
updated